src/utils/utils.ts

```
@@ -153,3 +156,119 @@ export const checkHasZipStructure = async (contentPath: string): Promise<boolean
 
   return true;
 };
+
+// build the file content in case of Link/App
+export const buildTextContent = (url: string, type: ItemType): string => {
+  if (type === ItemType.LINK) {
+    return `[InternetShortcut]\n${url}\n`;
+  }
+  return `[InternetShortcut]\n${url}\nAppURL=1\n`;
+};
+
+export const addItemToZip = async (args: {
+  item: Item;
+  archiveRootPath: string;
+  archive: Archiver;
+  member: Member;
+  fileServiceType: string;
+  iTM: ItemTaskManager;
+  runner: TaskRunner<Actor>;
+  fileTaskManager: FileTaskManager;
+  fileStorage: string;
+}) => {
+  const {
+    item,
+    archiveRootPath,
+    archive,
+    member,
+    fileServiceType,
+    iTM,
+    runner,
+    fileStorage,
+    fileTaskManager,
+  } = args;
+  // get item and its related data
+  const itemExtra = item.extra as Extra;
+
+  switch (item.type) {
+    case fileServiceType: {
+      let filepath = '';
+      let mimetype = '';
+      // check for service type and assign filepath, mimetype respectively
+      if (fileServiceType === FILE_ITEM_TYPES.S3) {
+        const s3Extra = item.extra as S3FileItemExtra;
+        ({ path: filepath, mimetype } = s3Extra.s3File);
+      } else {
+        const fileExtra = item.extra as LocalFileItemExtra;
+        ({ path: filepath, mimetype } = fileExtra.file);
+      }
+      // get file stream
+      const task = fileTaskManager.createDownloadFileTask(member, {
+        filepath,
+        itemId: item.id,
+        mimetype,
+        fileStorage,
+      });
+
+      const fileStream = (await runner.runSingle(task)) as ReadStream;
```

Maybe add a comment that these lines can throw if the file is not found.